src/utils/CacheManager.ts

```diff
@@ -0,0 +1,58 @@
+import NodeCache from "node-cache"
+
+export class CacheManager {
+  cachePools: Map<string, NodeCache> = new Map()
+
+  init({
+    pool,
+    ttl,
+    checkperiod,
+  }: {
+    pool: string
+    ttl?: number
+    checkperiod?: number
+  }) {
+    const cache = this.cachePools.get(pool)
+    if (cache) return
+    const newCache = new NodeCache({
+      stdTTL: ttl,
+      checkperiod,
+      useClones: false,
+    })
+    this.cachePools.set(pool, newCache)
+    return newCache
+  }
+
+  async get({
+    pool,
+    key,
+    call,
+    ttl = 120,
+  }: {
+    pool: string
+    key: string
+    call: () => Promise<any>
+    ttl?: number
+  }) {
+    const cache = this.cachePools.get(pool)
+    if (!cache) return {}
+    let val = cache.get(key)
+    if (!val) {
+      val = await call()
+      cache.set(key, val, ttl)
+    }
+    return val as any
+  }
+
+  private find(pool: string, str: string) {
+    const cache = this.cachePools.get(pool)
+    return cache?.keys().filter((k) => k.includes(str))
```

Comment:
chỗ `.includes` coi chừng về sau có khả năng tạo bug do nó match substring, an toàn hơn chút nhưng vẫn flexible thì nên dùng `startsWith`